src/Symfony/Component/Security/Core/User/User.php

```diff
@@ -18,7 +18,7 @@
  *
  * @author Fabien Potencier <fabien@symfony.com>
  */
-final class User implements AdvancedUserInterface
+final class User implements UserInterface, EquatableInterface
```

Comment:
Consider this instead;

`final class User implements UserInterface, EquatableInterface, AdvancedUserInterface`

It should work just fine : https://3v4l.org/eRO4P, and avoids a BC-break for code that does this: `$user instanceof AdvancedUserInterface`